### PR TITLE
Adds sentience event

### PR DESCRIPTION
**Pull request by SamHPurp**
Detects a random simple creature on the station... and turns them sentient. Yes, this includes The Best Officer :tm:

I was quite far past the ballmer peak as I ported this, so please review well.

All tests went well though.

Original design by @coiax, port to Paradise by me.

:cl: coiax & Purpose2
add: Adds a rare sentience event
/ :cl:

### CODE REVIEW

**Comment by Spartan6** [comment]
So essentially a simple animal turns into a walking talking thing?

Ian can suddenly invade the bridge with sentient corgi's?

**Comment by Fruerlund** [comment]
I do not dislike this but neither like it very much. Since it's rare it will generate fun scenarios once in a while.

**Comment by SamHPurp** [comment]
@Fruerlund aye, I made it very rare, so hopefully it is actually cool when it happens.

@Spartan6 updated my original message to actually describe what it does. Whoops.

File: code/modules/events/sentience.dm [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	SA.maxHealth = max(SA.maxHealth, 200)
+	SA.health = SA.maxHealth
+	SA.del_on_death = FALSE
+	greet_sentient(SA)
+
+/datum/event/sentience/proc/greet_sentient(var/mob/living/carbon/human/M)
+	to_chat(M, "<span class='userdanger'>Hello world!</span>")
+	to_chat(M, "<span class='warning'>Due to freak radiation, you have gained \
+	 						human level intelligence and the ability to speak and understand \
+							and the ability to speak and understand human language!</span>")
```
Comment (Citinited) [inline code/modules/events/sentience.dm]: Doesn't this line duplicate `and the ability to speak and understand`?

**Comment by TravellingMerchant** [comment]
Not sure I like it. Can the players playing the animal say no to it when it happens?

**Comment by Fox-McCloud** [comment]
Sentience potions explicitly don't work on bots, which are only "simple mobs" by virtue of saving on code and CPU cycles. For all intents and purposes, this event should include anything that you can't use a sentience potion on.

**Comment by SamHPurp** [comment]
@Fox-McCloud This does include Beepsky. Also someone becoming Beepsky is super duper rare, and are as easily disableable as usual via EMPs.

Typically a player is going to get a slime, mouse or monkey.

@TravellingMerchant This of course asks the ghosts if they wish to be a sentient creature... this doesn't automatically grab anyone that doesn't want to do it.

File: code/modules/events/sentience.dm (hunk 1 above)
Comment (SamHPurp) [inline code/modules/events/sentience.dm]: Whoops. I blame the Glenfiddich.

**Comment by TravellingMerchant** [comment]
What I mean is, what if someone is already playing as that creature? What happens then?

**Comment by FalseIncarnate** [comment]
I think Fox meant to say "this even't shouldn't include anything you can't use a sentience potion on." rather than "should".

Gonna agree with him on that too. Leave "sentient" bots to the realm of pAIs, and just make this check actual animals and such.

**Comment by SamHPurp** [comment]
@TravellingMerchant I already said, they don't even get considered for sentience.

@FalseIncarnate That does make this whole thing much much less interesting.

**Comment by Citinited** [comment]
This has got to be more interesting than wallrot, so 👍 to this from me

File: code/modules/events/sentience.dm [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	var/list/potential = list()
+	for(var/mob/living/simple_animal/L in living_mob_list)
+		var/turf/T = get_turf(L)
+		if (T.z != 1)
+			continue
+		if(!(L in player_list) && !L.mind)
+			potential += L
+
+	var/mob/living/simple_animal/SA = pick(potential)
+	var/mob/living/simple_animal/SG = pick(candidates)
```
Comment (FalseIncarnate) [inline code/modules/events/sentience.dm]: I suspect this isn't meant to be simple_animal for ghosts...

Comment (SamHPurp) [inline code/modules/events/sentience.dm]: eh... what would  you declare them as for a ghost? just `var/mob`?

**Comment by optimumtact** [comment]
I don't recall doing this, I think it was coiax

**Comment by TravellingMerchant** [comment]
In that case I'm all for it. Seems fun.

**Comment by SamHPurp** [comment]
Sorry @optimumtact , I must have looked at the wrong bit, think you merged it. I'll update.

**Comment by taukausanake** [comment]
This looks fun, but it doesn't look fun enough. As in anyone can possess Ian or any other animal on the station. This just gives the chance for one of them to speak to everyone. I rather it be more chaotic like any ghosts that possess gain the ability to speak, the rad storm gives all ghosts a chance to get in as a speaker. Having a rare event that only makes one thing able to speak sounds boring in my opinion

**Comment by fludd12** [comment]
@SamHPurp IIRC, /tg/ has it where the animal sentience event is separate from the simple robot sentience event, with the latter being much rarer. I suggest replicating that, since otherwise you have a player-controlled method of the rarity or otherwise of being selected for a sentient bot - either by killing off the animal population, or oversaturating the market with bots.

**Comment by taukausanake** [comment]
Maybe take away animal possession by ghosts, the option to respawn as one of them, just in favor of this. Means I can't be a little shit as Ian anymore but honestly this wouldn't do anything more for simple mobs

**Comment by SamHPurp** [comment]
@taukausanake this would cause so much rage.

@fludd12 I do not think this is accurate. I'll doublecheck though.

**Comment by fludd12** [comment]
Like I said, only if I recall correctly. Thanks for considering it, though.

**Comment by taukausanake** [comment]
@SamHPurp The only animals anyone controls anyways is Ian, Poly, or the captain's Fox; and the former two only is they want to be a shit. Barely anything would be missed if NPC possession was axed. But if you made it something better like what this is doing for a single random on the station then it might actually mean something. I for one would think it interesting for all the animals and bots on the station to gain sentience and see what it implies afterwards life-wise. That's just me though, I like seeing trouble on the station happen

**Comment by SamHPurp** [comment]
@fludd12 So, half right - yeah they have it split with the latest iteration of the code, but, it is no less common... but includes the possilbity of EVERYTHING gaining conciousness.

I'll have a look through their code and see if I think it is right for Paradise, or what tweaks might need to occur for our server... and it could be good to update it.

**Comment by SamHPurp** [comment]
@Fox-McCloud This only allows for Z1.

Enforcing this be only already sentience potioned on makes this an utterly utterly pointless thing.

If that is actually the requirement for making this merged, I'd rather close it than do it.

**Comment by Fox-McCloud** [comment]
After discussing this, yeah, we're going to want the sentience check; if you're not willing/don't want to do that, then just go ahead and close it; otherwise, if the changes are made, this will probably get merged.

**Comment by SamHPurp** [comment]
Sorry dudes, I won't delete my branch for a while in case someone else wants to take it - but I just don't see it adding to the gameplay with the proposed restrictions.

**Comment by Aurorablade** [comment]
I agree with the bots but it should extend to most simple animals.

**Comment by Necaladun** [comment]
I'd really prefer this done conservatively, and then we can look at expanding it to other things after we see how it is, and look at what exact mobs we'd allow this to work on.

If there's going to be no compromises, that's quite a shame.

**Comment by SamHPurp** [comment]
Neca, this isn't a rejection of all compromise... this is a, a clear restriction has been set, and I don't see this as an enjoyable event with that restriction in place.

It already differs from TGs implementation in that the sentience event effects EVERYTHING, meaning ALL ghosts could play something if they wanted.

Restricting this even further makes this even less interesting to the point where it triggering would likely just lead to less interesting events as it takes away from other events triggering in its place.

Maybe we can find more interesting ground - but with a hard restriction imposed, which I queried, and had confirmed (which seemed cordial to me) that I was no longer interested in this being a thing.

Thats going to happen a lot. Flattest's Space Pirate's PR being a prime example. Seemed cool, but the author decided he ultimately didn't like it.

Lets not get hung up on things like this, and trying to shame people - because ultimately its how you drive coders and contributors away - and our pool of coders is already severely limited compared to competing servers.

S

**Comment by Necaladun** [comment]
There's a lot of reasons why various mobs becoming sentient randomly could be a big problem, hence the desire to see it first in a lesser form, before extending it. Iron out any bugs or issues, then extend it to more dangerous things.

If that's not the way you want to work, that's fine, but it's the way we prefer to work here.

**Comment by SamHPurp** [comment]
Again, you're getting awfully sore over this. I did some work, it got neutered with no room for middle-ground. I don't feel the end result is valuable, I won't waste my time on it any further. I'm not upset over it, but you clearly sound it.

I don't mean for that to happen, but it has.

A whole event akin to three minutes work from a Xenobiologist does not a fun event make. Thus, its not something to bear completion if that is the restriction imposed.

Simple as that, and you attempt to attack me over this...

**Comment by Aurorablade** [comment]
there is middle ground: just let  animals get it,give them common to speak if they do not have it already (poly)..if you have to do a damage check on them and either make them do stamina damage or lower the damage.maybe hp buff them (like for mice)..and give them a fluff objective ("You now realize (area/job/object) is evil, avoid it at all costs" or "Collect all cable coils on station for a nest")...and you are good

**Comment by Spacemanspark** [comment]
Neca isn't 'attacking' you.
There was a compromise, a set of goals you could have made to accomplish this step by step, but you decided to close it off instead. Neca stated this to you quite nicely.
Don't accuse the head administrator trying to help your ass of "attacking" you.

**Comment by SamHPurp** [comment]
The offiical response, Fethas, was 'only animals that can be sentience potioned'. I questioned it. It got confirmed. So there isn't a proper middle ground, no.

And Spark, just 'lol'.

**Comment by Aurorablade** [comment]
You may wish to try it with my suggestion anyway and see what happens you may be surprised perhaps?

**Comment by SamHPurp** [comment]
That may be, but you cannot be surprised when my response to official guidance from the maintainers was to believe and listen to it, and not ignore it as you are suggesting.

**Comment by Aurorablade** [comment]
you would not be ignoreing it.

**Comment by Aurorablade** [comment]
okay i did a search on what the sentience potion actually hit..and its ALOT it hits mobs that are the sentinet_type of SENTIENT_ORGANIC https://github.com/ParadiseSS13/Paradise/search?utf8=%E2%9C%93&q=sentience_type&type=  theres a sentient type search there. most simple animals fall under sentient organic. so my idea is feasible and still can follow any guidlines setout.

**Comment by SamHPurp** [comment]
An event that literally is the same as a Xenobiologist tipping a sentient potion over a creature. Sure sounds exciting, and definitely worth taking up an event roll...

**Comment by Necaladun** [comment]
Neither trying to attack you nor awfully sore, apologies if it came off that way.

**Pull request closed without merging**